### DISCUSSION TITLE
회전 행렬의 직교 행렬로서의 특성

### DISCUSSION THREAD

**Discussion by Woon-2** [MyeongWoonJang/cpp-study#91] [General]
## 기저 변환

우리가 대학에서 배운 선형대수학은 생각보다 얕은 지식입니다.   
아름다운 수학적 표현들을 마다하고 무식하게 원소들을 일일이 나열해가며 정리들을 증명하고,   
그 어떤 수학적 의의도 제시하지 못한 수업이었죠.   
프로그래밍으로 치면 클래스와 함수를 이용하지 않고 `main`에 3000줄의 코드를 작성한 것이나 다름없습니다.

저 또한 이렇게 배웠을진데, 조금 더 찾아봤다고 해서 여러분에게 완벽히 수학적인 설명을 하는 건 불가능합니다.   

그래도 일단 직관에 의해서 설명을 해보려 할텐데,   
크게 의의를 두진 마세요.

<br>

먼저, 행렬곱에서 각 행과 각 열의 곱을 스칼라곱(내적)으로 생각해봅시다.   
열벡터 방식을 사용하고 있다고 가정하겠습니다.

<div align="center"><image src="https://user-images.githubusercontent.com/73771162/201558911-75fadc05-0c73-49cf-80a4-50c3937243a4.PNG"></div>
<div align="center"><image src="https://user-images.githubusercontent.com/73771162/201558913-4bb2e8de-ccbc-4b14-bf81-ef288dce1f75.PNG"></div>

세 표현이 완전히 동등함을 이해하셔야 합니다.

그리고 이를 기저변환으로 해석하는 것 역시 중요합니다.

**i**, **j**, **k**라는 기존의 기저를 **a**, **b**, **c**라는 새로운 기저로 변환한 것이죠.   

- 1열은 새로운 기저 **a**가 **i**, **j**, **k**를 이용해 어떻게 표현되는가를 나타냅니다.
- 2열은 새로운 기저 **b**가 **i**, **j**, **k**를 이용해 어떻게 표현되는가를 나타냅니다.
- 3열은 새로운 기저 **c**가 **i**, **j**, **k**를 이용해 어떻게 표현되는가를 나타냅니다.
- 1행은 기존의 기저 **i**가 **a**, **b**, **c**를 이용해 어떻게 표현되는가를 나타냅니다.
- 2행은 기존의 기저 **j**가 **a**, **b**, **c**를 이용해 어떻게 표현되는가를 나타냅니다.
- 3행은 기존의 기저 **k**가 **a**, **b**, **c**를 이용해 어떻게 표현되는가를 나타냅니다.

이것을 머릿속에 넣어두시고, 다음 글을 보시면 한 벡터를 다른 벡터를 이용해 표현한다는 것이 어떤 의미인지 짐작하실 수 있습니다.

[\[티스토리 Homo Robotics\] \[LINEAR ALGEBRA - 1\] \[선형대수학 이야기 - 1\] 기저와 기저변환](https://homo-robotics.tistory.com/16)

## 회전행렬은 왜 직교 행렬인가

왜 행렬곱이 기저변환으로 해석될 수 있단 걸 알아야 했을까요?   
기존 기저 벡터가 위의 **i**, **j**, **k**와 같이 나열했을 때 단위 행렬이 된다면,   
기저변환 행렬은 결국 새로운 기저 벡터들의 나열에 불과하기 때문입니다.

그리고 직교 행렬의 정의는,   

>그 행렬을 구성하는 열벡터 혹은 행벡터들이 정규 직교 기저를 이루는 행렬      

입니다.

여러분의 머릿속에 책상을 두어 보세요.   
그리고 그 위로 정육면체 주사위를 던집니다.   

주사위는 공중에서 여러 번 회전합니다.   
주사위가 회전한다고 해서 정육면체가 아니게 되진 않죠.   

정육면체의 모서리들은 항상 서로 수직입니다.   

주사위의 서로 수직인 모서리 세 개로 정규 직교 기저를 이룰 수 있죠.   
그 기저를 아무리 회전시켜도(변환시켜도) 여전히 정규 직교 기저입니다.

- 행렬곱은 기저변환으로 해석될 수 있습니다.
- 기존 기저 벡터를 나열했을 때 단위 행렬이 된다면, 기저변환 행렬은 새로운 기저 벡터들의 나열에 불과합니다.
- 그렇게 회전 행렬을 새로운 기저 벡터들의 나열로 해석할 수 있습니다.
- 기존의 기저가 정규 직교 벡터라면 아무리 회전시켜도 여전히 정규 직교 벡터입니다.
- 새로운 기저 벡터들이 정규 직교 벡터이므로, 회전 행렬은 직교 행렬입니다.

## 그래서, 직교행렬이라는 것엔 무슨 의미가 있는데?

간단합니다.   
첫 번째 항목에서 소개한 항등식을 잘 살펴보시면,   
직교 행렬 **Q**에 대해

<div align="center"><image src="https://user-images.githubusercontent.com/73771162/200862366-572f5767-8b69-484f-904e-5343d11c2995.PNG"></div>

가 성립한다는 것을 알아내실 수 있습니다.   
행렬곱을 벡터 스칼라곱의 나열로 생각할 수 있으니까요.   

아마 눈치 빠르시다면 알겠지만, **Q**는 역행렬이 곧 전치행렬입니다.   

위 식을 다르게 써보겠습니다.   
**Q**가 **e<sub>1</sub>**, **e<sub>2</sub>**, **e<sub>3</sub>** 라는 기저 벡터들의 나열이라고 생각하는 것이죠.

<div align="center"><image src="https://user-images.githubusercontent.com/73771162/201566293-1aa60255-0d9f-49d2-b109-0d05e970fad5.PNG"></div>
<div align="center"><image src="https://user-images.githubusercontent.com/73771162/201566584-2c13f255-afbe-4e90-8996-336e568e39f8.PNG"></div>

행렬곱의 결과는 단위행렬이 됩니다.

## 요약

기저변환이라는 점에서 회전행렬은 그저 새로운 기저 벡터들의 나열로 해석될 수 있고,   
직교행렬이라는 점에서 회전행렬은 역변환 시 전치행렬을 곱해주면 된다는 것을 알 수 있습니다.

**Comment by Lcat12345** [comment]
.1
---
![image](https://user-images.githubusercontent.com/62102830/201348039-cbbd2ca9-0ec7-4eb5-8e0b-792946d20818.png)

[\[티스토리 Homo Robotics\] \[LINEAR ALGEBRA - 1\] \[선형대수학 이야기 - 1\] 기저와 기저변환](https://homo-robotics.tistory.com/16)에서 이해가 안가는 부분인데요

열벡터 [1, 1]이 왜 a1[2, -1] + a2[-1, 1]이 되는지 모르겠습니다.

한 벡터를 다른 벡터를 이용해 표현한다는 의미는 알고 넘어갔습니다.

---

쓰다보니 별로 어려운건 없는것 같네요. 1에서 끝나버렷어요

**Reply by Woon-2** [reply]
기존의 기저 벡터가 **e1** = (1, 1)<sup>T</sup>, **e2** = (1, -1)<sup>T</sup>,   
새로운 기저 벡터가 **e'1** = (2, -1)<sup>T</sup>, **e'2** = (-1, 1)<sup>T</sup>인가 봅니다.   

기존의 기저 벡터를 새로운 기저 벡터로 표현하고 있네요.   

이용할 수 있는 식은

- **v** = 3 * **e1** + (-2) * **e2**
- **e1** = a<sub>1</sub> * **e'1** + a<sub>2</sub> * **e'2**
- **e2** = b<sub>1</sub> * **e'1** + b<sub>2</sub> * **e'2**

이고,

구해야 하는 것은 **v** = m * **e'1** + n * **e'2**의 (m, n)입니다.   

간단한 치환입니다.

**Reply by Lcat12345** [reply]
기저 벡터의 나열이 된다는 결과가 중요한 것 같습니다. 
아직은 대략적으로만 이해 된다는..
